src/library.js

```diff
@@ -1067,37 +1067,13 @@ LibraryManager.library = {
     return ret | 0;
   },
 
-  llvm_cttz_i32__deps: [function() {
-    function cttz(x) {
-      for (var i = 0; i < 8; i++) {
-        if (x & (1 << i)) {
-          return i;
-        }
-      }
-      return 8;
-    }
-    if (SIDE_MODULE) return ''; // uses it from the parent
-
-#if USE_PTHREADS
-    return 'var cttz_i8; if (ENVIRONMENT_IS_PTHREAD) cttz_i8 = PthreadWorkerInit.cttz_i8; else PthreadWorkerInit.cttz_i8 = cttz_i8 = allocate([' + range(256).map(function(x) { return cttz(x) }).join(',') + '], "i8", ALLOC_STATIC);';
-#else
-    return 'var cttz_i8 = allocate([' + range(256).map(function(x) { return cttz(x) }).join(',') + '], "i8", ALLOC_STATIC);';
-#endif
-  }],
 #if WASM == 0 // binaryen will convert these calls to wasm anyhow
   llvm_cttz_i32__asm: true,
 #endif
   llvm_cttz_i32__sig: 'ii',
   llvm_cttz_i32: function(x) {
-    x = x|0;
-    var ret = 0;
-    ret = {{{ makeGetValueAsm('cttz_i8', 'x & 0xff', 'i8') }}};
-    if ((ret|0) < 8) return ret|0;
-    ret = {{{ makeGetValueAsm('cttz_i8', '(x >> 8)&0xff', 'i8') }}};
-    if ((ret|0) < 8) return (ret + 8)|0;
-    ret = {{{ makeGetValueAsm('cttz_i8', '(x >> 16)&0xff', 'i8') }}};
-    if ((ret|0) < 8) return (ret + 16)|0;
-    return ({{{ makeGetValueAsm('cttz_i8', 'x >>> 24', 'i8') }}} + 24)|0;
+    x = x | 0;
+    return (x ? 31 - Math_clz32(x ^ x - 1) | 0 : 32) | 0; 
```

Comment:
is there a quick proof for why this is correct?

Reply:
A number `x > 0` can be written down in binary in form:

```
x=aaa...aaa 1 000...000
```

where `aaa...aaa` is a high part, and the `1` is the position of the lowest bit that is set, and all bits lower than that are zero. Here since `x` is 32-bit
```
len(aaa...aaa) + len(1) + len(000...000) = 32
```
and by definition `cttz(x) = len(000...000)`, so
```
cttz(x) = 32 - len(aaa...aaa) - 1
```

So if we can find `len(aaa...aaa)`, we then know `cttz(x)`.

Computing `x - 1` has the effect of zeroing the lowest bit that was set, and setting all the bits lower than that to one, i.e.:

```
x-1 = aaa...aaa 0 111...111
```
and therefore
```
        x = aaa...aaa 1 000...000
      x-1 = aaa...aaa 0 111...111
 --------------------------------
x ^ (x-1) = 000...000 1 111...111
```

where it is seen that `len(000...000)` in this resulting number is exactly same as `len(aaa...aaa)` above. That can be counted by counting leading zeroes, so `len(aaa...aaa) = clz(x ^ (x-1))`, and finally

```
cttz(x) = 32 - len(aaa...aaa) - 1 = 31 - clz(x ^ (x-1))
```